### ISSUE TITLE
Display only uncorrectable offenses

### ISSUE THREAD

**Issue by bogdan** [rubocop/rubocop#13275]
## Is your feature request related to a problem? Please describe.

In my project, most offences are automatically corrected by github action. I need to display to a developer only offences that are not automatically correctable without actually fixing them because I don't want to let developer know about things he doesn't need to know.

## Describe the solution you'd like

Add `--display-only-uncorrectable` console option that would only display offences that can not be corrected.

## Describe alternatives you've considered

I tried `--format=json` and filter the offences manually via `files[].offences[].correctable` but it requires too much formatting code on top.  It is possible though to implement `--filter` option that would allow for some flexible filtering like [jq](https://jqlang.github.io/jq/manual/)

**Comment by Earlopain** [comment]
This seems pretty specific. Could you instead do with a custom formatter?

```rb
class CustomFormatter < RuboCop::Formatter::ClangStyleFormatter
  def file_finished(file, offenses)
    offenses = offenses.reject(&:correctable?) unless ENV["CI"]
    super
  end
end
```

Adding `--format CustomFormatter` to `.rubocop` will then automatically use it. You also need to add that file to the `require` section of `.rubocop.yml` for it to work.

**Comment by bogdan** [comment]
The following solved my problem:

``` ruby
# frozen_string_literal: true

require 'rubocop/formatter/json_formatter'
require 'rubocop/formatter/clang_style_formatter'
require 'rubocop/formatter/progress_formatter'

module SkipCorrectableRubocopFormatter
  def file_finished(file, offenses)
    offenses = offenses.reject(&:correctable?)
    super
  end
end

[
  RuboCop::Formatter::JSONFormatter,
  RuboCop::Formatter::ClangStyleFormatter,
  RuboCop::Formatter::ProgressFormatter,
].each do |klass|
  klass.send(:prepend, SkipCorrectableRubocopFormatter)
end
```

Making sure all formatters used in certain env do not output the correctable offences.
I would say the better idea would be to have `--filter CustomOffencesFilter` separated from `--format`. But I will leave it for core team to decide.

**Comment by Earlopain** [comment]
Yeah, it isn't as nice as it could be. If something does get added, I would personally go with `--display-only-uncorrectable` instead of something new alltogether.

Perhaps all the other `--display-only-` can be consolidated under a single switch like `--display-only` so that you can write `--display-only=correctable`/`--display-only=uncorrectable`,etc. (not sure, I think all these should be exclusive to each other). It's getting a bit much.

**Comment by Zopolis4** [comment]
I'd also like this for my rubocop-crystal extension, as outlined in https://github.com/Zopolis4/rubocop-crystal/commit/99fb73e8bc6417e1aa19b5a26d1e882335338074.
